src/object_storage/s3.rs

```diff
@@ -60,48 +169,138 @@ impl ObjectStorage for S3 {
     }
 
     fn create_bucket(&self, bucket_name: &str) -> Result<(), EngineError> {
-        let mut cmd = QoveryCommand::new(
-            "aws",
-            &vec!["s3api", "create-bucket", "--bucket", bucket_name],
-            &self.credentials_environment_variables(),
-        );
-        cast_simple_error_to_engine_error(
-            self.engine_error_scope(),
-            self.context().execution_id(),
-            cmd.exec()
-                .map_err(|err| SimpleError::new(Other, Some(format!("{}", err)))),
-        )
+        if let Err(message) = S3::is_bucket_name_valid(bucket_name) {
+            let message = format!(
+                "While trying to create S3 bucket, name `{}` is invalid: {}",
+                bucket_name,
+                message.unwrap_or_else(|| "unknown error".to_string())
+            );
+            return Err(self.engine_error(EngineErrorCause::Internal, message));
+        }
+
+        let s3_client = self.get_s3_client();
+
+        // check if bucket already exists, if so, no need to recreate it
+        if self.bucket_exists(bucket_name) {
+            return Ok(());
+        }
+
+        if let Err(e) = block_on(s3_client.create_bucket(CreateBucketRequest {
+            bucket: bucket_name.to_string(),
+            create_bucket_configuration: Some(CreateBucketConfiguration {
+                location_constraint: Some(self.region.to_aws_format()),
+            }),
+            ..Default::default()
+        })) {
+            let message = format!(
+                "While trying to create S3 bucket, name `{}` region `{}`: {}",
+                bucket_name,
+                self.region.to_aws_format(),
+                e
+            );
+            error!("{}", message);
+            return Err(self.engine_error(EngineErrorCause::Internal, message));
+        }
+
+        let creation_date: DateTime<Utc> = Utc::now();
+        if let Err(e) = block_on(s3_client.put_bucket_tagging(PutBucketTaggingRequest {
+            bucket: bucket_name.to_string(),
+            expected_bucket_owner: None,
+            tagging: Tagging {
+                tag_set: vec![
+                    Tag {
+                        key: "CreationDate".to_string(),
+                        value: format!("{}", creation_date.to_rfc3339()),
+                    },
+                    Tag {
```

Comment:
I prefer not to see this. Please prepare the request before, and add the TTL if necessary. If we run Pleco at some points, 0 means immediatly, which is dangerous

Reply:
Done.